auth.c

```diff
@@ -53,13 +68,95 @@ void janus_auth_deinit(void) {
 	if(allowed_plugins != NULL)
 		g_hash_table_destroy(allowed_plugins);
 	allowed_plugins = NULL;
+	if(auth_secret != NULL)
+		g_free(auth_secret);
+	auth_secret = NULL;
 	janus_mutex_unlock(&mutex);
 }
 
+gboolean janus_auth_check_signature(const char *token, const char *realm) {
+	if (!auth_enabled || auth_secret == NULL)
+		return FALSE;
+	gchar *token_signature = strchr(token, ':') + 1;
+	gchar *token_metadata = g_strndup(token, token_signature - token - 1);
+	gchar **list = g_strsplit(token_metadata, ",", 3);
```

Comment:
The lines above look dangerous... they seem to give for granted that a colon will be in the string, and doing math on those to extract pointers, which can lead to crashes or overflows/underflows if that's not the case.

Reply:
you are right, i relaxed my checks too much here while refactoring. Originally I did a `g_strsplit` here too, but it seemed like a simple enough thing to optimize.

will checking the `strchr` result be enough or would you like to get rid of the pointer math completely?

Reply:
I think that doing another `g_strsplit` shouldn't impact much, as even now you're doing an allocation anyway: besides, I guess `g_strsplit` is written in a way that does all those checks properly for you. I guess that `sscanf` can't be used, here (no spaces). Maybe regular expressions? (I'd have to check if glib has those, though, I wouldn't want an additional dependency just for that).

Reply:
@lminiero sscanf should work with colons too, no? `sscanf("%...s:%s",data,token)`. The main problem is that I have to choose a buffer size and enforce it then too etc - `g_strsplit` is fine, even though all the array indexing is a bit ugly. Regex are too much for this IMO.

Reply:
It should be possible to do that with sscanf, but [not by just putting the colon in there](https://stackoverflow.com/questions/28387564/trouble-with-delimiting-colon-with-scanf). Anyway, I agree regex is overkill and g_strsplit is fine.